usr/src/cmd/mdb/common/mdb/mdb_cmds.c

```diff
@@ -198,6 +198,100 @@ write_uint64(mdb_tgt_as_t as, mdb_tgt_addr_t addr, uint64_t n, uint_t rdback)
 	return (addr + sizeof (n));
 }
 
+/*
+ * Writes to objects of size 1, 2, 4, or 8 bytes. The function
+ * doesn't care if the object is a number or not (e.g. it could
+ * be a byte array, or a struct) as long as the size of the write
+ * is one of the aforementioned ones.
+ */
+static mdb_tgt_addr_t
+write_var_uint(mdb_tgt_as_t as, mdb_tgt_addr_t addr, uint64_t val, size_t size,
+    uint_t rdback)
+{
+	if (size > sizeof (uintptr_t)) {
```

Comment:
Why are we using the uintptr_t as the sentinel for the size? For example, if we were building a 32-bit version of this, then we'd have an ILP32 system so the uintptr_t would be 4 bytes; however, you can write 8 bytes.

Reply:
You are right. The check here is to ensure that we don't write more bytes than the longest integer supported.

Looking into the different integer types it seems like `uintmax_t` would be the right type for this. Does this sound reasonable to you?

Reply:
It might be, but even that is a little questionable. The max we have in the code is 8 and we're not relating that to a type. So I'd probably actually use that and just make a `#define` with a comment that explains its the max size that the targets can write today. As it's really less specific to a type and more specific to what the target allows.

Reply:
Makes sense. I'll update the commit. Thank you for reviewing this.